icu4j/main/core/src/main/java/com/ibm/icu/segmenter/RuleBasedSegmenter.java

```diff
@@ -0,0 +1,137 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.segmenter;
+
+import com.ibm.icu.text.BreakIterator;
+import com.ibm.icu.text.RuleBasedBreakIterator;
+import java.io.InputStream;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Performs segmentation according to the provided rule string. The rule string must follow the
+ * same guidelines as for {@link RuleBasedBreakIterator#getInstanceFromCompiledRules(InputStream)}.
+ * @draft ICU 78
+ */
+public class RuleBasedSegmenter implements Segmenter {
+
+  private String rules;
+
+  /**
+   * @draft ICU 78
+   */
+  @Override
+  public Segments segment(CharSequence s) {
+    return new RuleBasedSegments(s, this);
+  }
+
+  /**
+   * @return a builder for constructing {@code RuleBasedSegmenter}
+   * @draft ICU 78
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  RuleBasedSegmenter(String rules) {
+    this.rules = rules;
+  }
+
+  /**
+   * @internal
+   * @deprecated This API is ICU internal only.
+   */
+  @Override
+  @Deprecated
+  public BreakIterator getNewBreakIterator() {
```

Comment:
Same thoughts as the version of this method on `LocalizedSegmenter`.  I think you could avoid having this be public by either having the segmenter pass the break iterator to the `Segments` object on construction or by having this actually be a method on the `Segments` object.

Reply:
See conversation in [thread above](https://github.com/unicode-org/icu/pull/3237/files#r2049615459).

Reply:
Done. Ditto.